### ISSUE TITLE
Notify and Listen components throw error: object has no attribute 'build_results' + incompatible component inputs

### ISSUE THREAD

**Issue by marfal** [langflow-ai/langflow#4322]
### Bug Description

Notify and Listen components have Data and Text inputs that are incompatible with other nodes. Additionally both components throw error:

Error building Component Notify: 

'NotifyComponent' object has no attribute 'build_results'

Traceback (most recent call last):
  File "/app/.venv/lib/python3.12/site-packages/langflow/graph/vertex/base.py", line 727, in _build_results
    result = await initialize.loading.get_instance_results(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/langflow/interface/initialize/loading.py", line 67, in get_instance_results
    return await build_component(params=custom_params, custom_component=custom_component)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/langflow/interface/initialize/loading.py", line 153, in build_component
    build_results, artifacts = await custom_component.build_results()
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NotifyComponent' object has no attribute 'build_results'

And similarly:
Error building Component Listen: 

'ListenComponent' object has no attribute 'build_results'

Traceback (most recent call last):
  File "/app/.venv/lib/python3.12/site-packages/langflow/graph/vertex/base.py", line 727, in _build_results
    result = await initialize.loading.get_instance_results(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/langflow/interface/initialize/loading.py", line 67, in get_instance_results
    return await build_component(params=custom_params, custom_component=custom_component)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/langflow/interface/initialize/loading.py", line 153, in build_component
    build_results, artifacts = await custom_component.build_results()
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'ListenComponent' object has no attribute 'build_results'

Rendering them unusable. 

### Reproduction

1. Add component Notify and/or Listen on the canvas.
2. Select Data input handle or Name input handle
3. See the available list of available inputs / connective components
4. Try to build/run Notify and/or Listen components  
 

### Expected behavior

Expect input Data handle and Text handle to connect with correct inputs/outputs and for Notify and Listen components to execute correctly. 

### Who can help?

@ogabrielluiz @nicoloboschi

### Operating System

Ubuntu Linux 22.04

### Langflow Version

v1.0.19.post1

### Python Version

None

### Screenshot

![image](https://github.com/user-attachments/assets/0529977d-b8a2-46b4-834c-292a60ceb8b6)
![image](https://github.com/user-attachments/assets/80005100-0932-4ca1-bfec-be05e03dd720)
![image](https://github.com/user-attachments/assets/572ad918-fc94-49ef-b53e-b48fce503787)

**Comment by guoguoguilai** [comment]
Any idea to fix it?

**Comment by marfal** [comment]
Patch up the connectors is fairly straightforward but it seems the componentes are incomplete otherwise, and even after trying to implement the build_result I was never able to get them working correctly (probably due to my limited skills and knowledge of Langflow).

**Comment by joaoguilhermeS** [comment]
Hello @marfal @guoguoguilai the components were in an old format, they are still going to be reformulated that's why they are in beta. I have updated both of them.

[Listen.json](https://github.com/user-attachments/files/17622483/Listen.json)
[Notify.json](https://github.com/user-attachments/files/17622484/Notify.1.json)

**Comment by marfal** [comment]
@joaoguilhermeS Hi. I appreciate your help, but I guess these components are just not ready from the prime time yet. While the connector handles seem to be of a correct type now, there is still an error when trying to actually build/run the components. For example notify gies this error:
Error building Component Notify NEW: 

cannot access local variable 'data' where it is not associated with a value

Traceback (most recent call last):
  File "/app/.venv/lib/python3.12/site-packages/langflow/graph/vertex/base.py", line 727, in _build_results
    result = await initialize.loading.get_instance_results(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/langflow/interface/initialize/loading.py", line 67, in get_instance_results
    return await build_component(params=custom_params, custom_component=custom_component)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/langflow/interface/initialize/loading.py", line 153, in build_component
    build_results, artifacts = await custom_component.build_results()
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/langflow/custom/custom_component/component.py", line 707, in build_results
    return await self._build_with_tracing()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/langflow/custom/custom_component/component.py", line 695, in _build_with_tracing
    _results, _artifacts = await self._build_results()
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/langflow/custom/custom_component/component.py", line 731, in _build_results
    result = method()
             ^^^^^^^^
  File "<string>", line 32, in build_notify
UnboundLocalError: cannot access local variable 'data' where it is not associated with a value

**Comment by joaoguilhermeS** [comment]
Hey @marfal , although I didn't initially encounter the problem, I was able to replicate and fix it along with some other minor issues. I believe everything is now fully resolved and will be opening a PR soon to definitively address it. In the meantime, you can use this flow: [Notify and Listen - Example.json](https://github.com/user-attachments/files/17636453/Notify.and.Listen.-.Example.1.json).

**Comment by SnakeO** [comment]
@joaoguilhermeS Maybe I'm a bit confused -- I thought the purpose of notify/listen is that you don't need to have them connected and they would be "connected" via the event name. So that if the notifier fires the event, then the listener automatically picks it up and runs. For example:

![image](https://github.com/user-attachments/assets/d3f20c30-49dd-4c68-b936-da7211825ff1)

Will this not work?

**Comment by joaoguilhermeS** [comment]
@SnakeO, you are right, this is still ongoing work and might be fixed soon. Feel free to contribute to the PR as well.

**Comment by SnakeO** [comment]
Thank you @joaoguilhermeS -- can you explain in plain english how this might be accomplished without having an input connected to the listener, and I'll see if I can get it working?

**Comment by mukundha** [comment]
+1 seeing the same error, any plans to fix this

also kindly clarify if this fix will make `notify -> listen` work without explicitly connecting them

**Comment by puppetm4st3r** [comment]
any news on this ?
